### PR TITLE
6pm b track logins

### PR DESCRIPTION
**Pull request by matthewwong212**
mw - Frontend from issue #34 and backend from issue #33 of logins function implemented.

On the front end, admins can view a list of logins to the app, with timestamp, email, and first/last name of each user.

On the back end, a MongoDB database stores the table of logins.  The user information and corresponding timestamp is logged every time `isMember` or `isAdmin` is called in `Auth0MembershipService.java`

### CODE REVIEW

File: javascript/src/test/components/Admin/LoginTable.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    test("renders without crashing on empty list", () => {
+        render(<LoginTable loginTable={[]} />);
+    });
+
+
+  test("renders without crashing on non empty list", () => {
+
+    const loginTable = [
+        {
+            "timestamp": "7:00",
```
Comment (kylekam) [inline javascript/src/test/components/Admin/LoginTable.test.js:16]: Timestamp data looks slightly different in mongodb collections but I don't think it'll impact anything.
```suggestion
            "timestamp": "2020-12-08 14:41:48.481 -0800",
```

File: javascript/src/test/pages/Admin/ViewLogins.test.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+jest.mock("@auth0/auth0-react");
+
+import { fetchWithToken } from "main/utils/fetch";
+jest.mock("main/utils/fetch");
+
+describe("ChannelList tests", () => {
+  
+      const logins = [
+        {
+            "timestamp": "8:00",
```
Comment (kylekam) [inline javascript/src/test/pages/Admin/ViewLogins.test.js:18]: timestamp
```suggestion
            "timestamp": "2020-12-08 14:41:48.481 -0800",
```

File: javascript/src/test/pages/Admin/ViewLogins.test.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    }
+  
+    test("renders without crashing", () => {
+        useSWR.mockReturnValue({});
+        render(<ViewLogins />);
+    });
+    
+    test("Accesses logins from backend", () => {
+        const exampleLogin = {
+            "timestamp": "8:00",
```
Comment (kylekam) [inline javascript/src/test/pages/Admin/ViewLogins.test.js:49]: timestamp
```suggestion
            "timestamp": "2020-12-08 14:41:48.481 -0800",
```

File: javascript/src/test/pages/Admin/ViewLogins.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            "email": "test@ucsb.edu",
+            "firstName": "matthew",
+            "lastName": "wong"
+        };
+
+        useSWR.mockReturnValue({
+            'data': [exampleLogin]
+        });
+        const { getByText } = render(<ViewLogins />);
+        const timestampElement = getByText(/8:00/);
```
Comment (kylekam) [inline javascript/src/test/pages/Admin/ViewLogins.test.js:59]: timestamp
```suggestion
        const timestampElement = getByText(/2020-12-08 14:41:48.481 -0800/);
```

**Review by zhengyingl — COMMENTED** [review]
Looks good

File: src/main/java/edu/ucsb/ucsbcslas/documents/Login.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+/*
+Example of the JSON for a Login document:
+
+{
+    "_id": {
+        "$oid": "5fcf123e546a77556efb8345"
+    },
+    "timestamp": "10:00",
```
Comment (kylekam) [inline src/main/java/edu/ucsb/ucsbcslas/documents/Login.java:98]: timestamp
```suggestion
    "timestamp": "2020-12-08 14:41:48.481 -0800",
```

**Review by kylekam — DISMISSED** [review]
Only thing was that the timestamp data looks slightly different in the mongodb database than in the tests, but it shouldn't impact anything

**Review by zhengyingl — APPROVED** [review]

**Review by mara-downing — CHANGES_REQUESTED** [review]
Everything in the code looks good.

However, the user's role (admin/member) does not appear properly either on Heroku or localhost, which should be fixed before this branch is ready to be merged into main, especially since having an admin role is necessary to be able to test the functionality of this PR.

**Comment by matthewwong212** [comment]
> 
> 
> Everything in the code looks good.
> 
> However, the user's role (admin/member) does not appear properly either on Heroku or localhost, which should be fixed before this branch is ready to be merged into main, especially since having an admin role is necessary to be able to test the functionality of this PR.

It appears to be working for me on localhost.  I've also fixed it on Heroku, though only Prof Conrad's email, and my email are registered as Admins

**Comment by mara-downing** [comment]
> > Everything in the code looks good.
> > However, the user's role (admin/member) does not appear properly either on Heroku or localhost, which should be fixed before this branch is ready to be merged into main, especially since having an admin role is necessary to be able to test the functionality of this PR.
> 
> It appears to be working for me on localhost. I've also fixed it on Heroku, though only Prof Conrad's email, and my email are registered as Admins

It seems to work for me on Heroku now, thanks!

**Review by mara-downing — APPROVED** [review]
LGTM

**Comment by pconrad** [comment]
40 points!

**Pull request merged**